libqtile/utils.py

```diff
@@ -310,3 +314,84 @@ def scan_files(dirpath, *names):
         files[name].extend(found)
 
     return files
+
+
+async def _send_dbus_message(callback=None, session_bus=False, signal_name=None,
+                             dbus_interface=None, bus_name=None, path=None,
+                             remove=False):
+
+    match_args = {
+        "type": "signal",
+        "sender": bus_name,
+        "member": signal_name,
+        "path": path,
+        "interface": dbus_interface
+    }
+
+    rule = ",".join("{}='{}'".format(k, v)
+                    for k, v in match_args.items() if v)
+
+    if session_bus:
+        bus_type = BusType.SESSION
+    else:
+        bus_type = BusType.SYSTEM
+
+    bus = await MessageBus(bus_type=bus_type).connect()
+
+    member = "AddMatch" if not remove else "RemoveMatch"
+
+    subscribe = await bus.call(
+        Message(destination='org.freedesktop.DBus',
+                interface='org.freedesktop.DBus',
+                path='/org/freedesktop/DBus',
+                member=member,
+                signature='s',
+                body=[rule]))
+
+    # Check if message sent successfully
+    if subscribe.message_type == MessageType.METHOD_RETURN:
+        if not remove:
+            bus.add_message_handler(callback)
+        return True
+    else:
+        return False
+
+
+async def add_signal_handler(callback, session_bus=False, signal_name=None,
+                             dbus_interface=None, bus_name=None, path=None):
+    """
+    Helper function which aims to recreate python-dbus's add_signal_handler
+    method in dbus_next with asyncio calls.
+
+    Subscriptions can be removed with remove_signal_handler.
+
+    Returns True if subscription is successful.
+    """
+    success = await _send_dbus_message(callback=callback,
+                                       session_bus=session_bus,
+                                       signal_name=signal_name,
+                                       dbus_interface=dbus_interface,
+                                       bus_name=bus_name,
+                                       path=path)
+
+    return success
+
+
+async def remove_signal_handler(session_bus=False, signal_name=None,
+                                dbus_interface=None, bus_name=None, path=None):
```

Comment:
Looks like it isn't used anywhere?

Reply:
Correct. It's not currently used but if we're making a helper function to add the handler I thought it appropriate to provide a mechanism to remove it. python-dbus has `add_signal_receiver` and `remove_signal_receiver`.

The mistake I've made here is that I do need to pass the callback function as this needs to be removed too.

However, if you would prefer not to include it here, I can just delete it.

Reply:
Yeah, don't include dead code, we'll add it later if needed.

Reply:
No problem. Will strip out.